source/fundamentals/connection/connection-options.txt

```diff
@@ -0,0 +1,414 @@
+       | **Default**: ``null``
+
+   * - **ClusterConfigurator**
+     - {+not-available+}
+     - | Low-level configuration options for sockets, TLS, cluster, and others. 
+       | 
+       | **Data Type**: Action<`ClusterBuilder <{+api-root+}/T_MongoDB_Driver_Core_Configuration_ClusterBuilder.htm>`__>
+       | **Default**: ``null``
+
+   * - **Compressors**
+     - ``compressors=snappy,zstd``
+     - | The preferred compression types, in order, for wire-protocol messages sent to 
+         or received from the server. The driver uses the first of these compression types 
+         that the server supports. See :ref:`csharp-network-compression` for 
+         more information. 
+       | 
+       | **Data Type**: `CompressorConfiguration <{+api-root+}/T_MongoDB_Driver_Core_Configuration_CompressorConfiguration.htm>`__
+       | **Default**: ``null``
+
+   * - **ConnectTimeout**
+     - ``connectTimeoutMS=0``
+     - | The length of time the driver tries to establish a single TCP socket connection 
+         to the server before timing out. Specifying ``0`` disables the connection timeout.
+       |
+       | **DataType**: ``TimeSpan``
+       | **Default**: 30 seconds
+
+   * - **Credential**
+     - .. code-block:: none
+        :copyable: false
+        :source: none
+     
+        mongodb://user1:password1
+        &authMechanism=GSSAPI
+        &authMechanismProperties=SERVICE_NAME:other,REALM:otherrealm
+        &authSource=$external
+     - | Settings for how the driver authenticates to the server. This includes 
+         authentication credentials, mechanism, source, and other settings.
+       |
+       | If you don't specify an authentication mechanism, the driver uses either 
+       | ``SCRAM-SHA-1`` or ``SCRAM-SHA-256``, depending on the server version. See
+       | :ref:`authentication mechanisms <csharp-authentication-mechanisms>` for available
+       | authentication mechanisms.
+       |
+       | **Data Type**: `MongoCredential <{+api-root+}/T_MongoDB_Driver_MongoCredential.htm>`__
+       | **Default**: ``null``
+
+   * - **DirectConnection**
+     - ``directConnection=true``
+     - | Specifies whether to force dispatch **all** operations to the host.
+       |
+       | If you specify this option, the driver doesn't accept the 
+       | :manual:`DNS seed list connection format. <reference/connection-string/#std-label-connections-dns-seedlist>`
+       | You must use the :manual:`standard connection URI format <reference/connection-string/#standard-connection-string-format/>` 
+       | instead. 
+       | **Data Type**: {+bool-data-type+}
+       | **Default**: ``false``
+
+   * - **HeartbeatInterval**
+     - ``heartbeatFrequencyMS=5000``
+     - | The interval between regular server-monitoring checks. Must be greater than or 
+         equal to 500 milliseconds.
+       |
+       | **Data Type**: ``TimeSpan``
+       | **Default**: 10 seconds
+
+   * - **HeartbeatTimeout**
+     - {+not-available+}
```

Comment:
it's possible to configure it via connection [string](https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1008)